tracking/include/pcl/tracking/impl/approx_nearest_pair_point_cloud_coherence.hpp

```diff
@@ -13,11 +13,11 @@ ApproxNearestPairPointCloudCoherence<PointInT>::computeCoherence(
 {
   double val = 0.0;
   // for (std::size_t i = 0; i < indices->size (); i++)
-  for (std::size_t i = 0; i < cloud->points.size(); i++) {
+  for (const auto& point : *cloud) {
     int k_index = 0;
     float k_distance = 0.0;
     // PointInT input_point = cloud->points[(*indices)[i]];
-    PointInT input_point = cloud->points[i];
+    PointInT input_point = point;
```

Comment:
This can be declared in the for-range loop as a copy, in case you it needs to be one. I actually suspect that we don't need to make this copy.